x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Expressions.java

```diff
@@ -126,7 +121,12 @@ public static TypeResolution typeMustBe(Expression e, Predicate<Expression> pred
     }
 
     public static TypeResolution typeMustBeNumeric(Expression e) {
-        return e.dataType().isNumeric()? TypeResolution.TYPE_RESOLVED : new TypeResolution(
-                "Argument required to be numeric ('" + Expressions.name(e) + "' of type '" + e.dataType().esType + "')");
+        return e.dataType().isNumeric() ?
+            TypeResolution.TYPE_RESOLVED :
+            new TypeResolution(getErrorMessageForNumericRequirement(e));
+    }
+
+    public static String getErrorMessageForNumericRequirement(Expression e) {
```

Comment:
Minor nitpick to remove the get prefix (as this is not a getter) - maybe something like `numericErrorMessage`?